tests/blast/test_blast_statistics.h

```diff
@@ -0,0 +1,280 @@
+
+#ifndef SEQAN_TESTS_TEST_BLAST_STATISTICS_H_
+#define SEQAN_TESTS_TEST_BLAST_STATISTICS_H_
+
+#include <seqan/basic.h>
+#include <seqan/sequence.h>
+
+#include <seqan/blast.h>
+
+using namespace seqan;
+
+SEQAN_DEFINE_TEST(test_blast_scoring_scheme_conversion)
+{
+    typedef Blosum62 TScheme;
+    BlastScoringScheme<TScheme> scheme;
+    setScoreGapExtend(scheme, -1);
+    setScoreGapOpenBlast(scheme, -11);
+    SEQAN_ASSERT(isValid(scheme));
+
+    SEQAN_ASSERT_EQ(scoreGapExtend(scheme), -1);
+    SEQAN_ASSERT_EQ(scoreGapOpenBlast(scheme), -11);
+    SEQAN_ASSERT_EQ(scoreGapOpen(scheme), -12);
+
+    SEQAN_ASSERT_EQ(scoreGapExtend(seqanScheme(scheme)), -1);
+    SEQAN_ASSERT_EQ(scoreGapOpen(seqanScheme(scheme)), -12);
+
+    setScoreGapExtend(scheme, -3);
+    SEQAN_ASSERT(!isValid(scheme)); // no karlin altschul values exist for this combination
+    SEQAN_ASSERT_EQ(scoreGapExtend(scheme), -3);
+    SEQAN_ASSERT_EQ(scoreGapOpenBlast(scheme), -11);
+    SEQAN_ASSERT_EQ(scoreGapOpen(scheme), -14); // open changed as well!
+
+    SEQAN_ASSERT_EQ(scoreGapExtend(seqanScheme(scheme)), -3);
+    SEQAN_ASSERT_EQ(scoreGapOpen(seqanScheme(scheme)), -14);
+
+    setScoreGapExtend(scheme, -2);
+    setScoreGapOpen(scheme, -9); // setScorGapOpen in seqan convention
+    SEQAN_ASSERT(isValid(scheme)); // valid again
+    SEQAN_ASSERT_EQ(scoreGapExtend(scheme), -2);
+    SEQAN_ASSERT_EQ(scoreGapOpenBlast(scheme), -7);
+    SEQAN_ASSERT_EQ(scoreGapOpen(scheme), -9); // open changed as well!
+
+    SEQAN_ASSERT_EQ(scoreGapExtend(seqanScheme(scheme)), -2);
+    SEQAN_ASSERT_EQ(scoreGapOpen(seqanScheme(scheme)), -9);
+}
+
+SEQAN_DEFINE_TEST(test_blast_scoring_scheme)
+{
+    // Blosum62 and some general stuff
+    {
+        typedef Blosum62 TScheme;
+        BlastScoringScheme<TScheme> scheme;
+        setScoreGapOpenBlast(scheme, -11);
+        setScoreGapExtend(scheme, -1);
+
+        // TEST isValid()
+        SEQAN_ASSERT(isValid(scheme));
+        // TEST get*
+        SEQAN_ASSERT_EQ(getLambda(scheme), 0.267);
+        SEQAN_ASSERT_EQ(getKappa(scheme),  0.041);
+        SEQAN_ASSERT_EQ(getH(scheme),      0.140);
+        SEQAN_ASSERT_EQ(getAlpha(scheme),  1.900);
+        SEQAN_ASSERT_EQ(getBeta(scheme), -30.000);
+
+        setScoreGapOpen(scheme, -1);
+        setScoreGapExtend(scheme, -11);
+        SEQAN_ASSERT(!isValid(scheme));// invalid parameters, because not the setScoreGap*Blast version
+
+        // reset to valid
+        setScoreGapOpenBlast(scheme, -11);
+        setScoreGapExtend(scheme, -1);
+
+        // now check values again
+        SEQAN_ASSERT(isValid(scheme));
+        // TEST get*
+        SEQAN_ASSERT_EQ(getLambda(scheme), 0.267);
+        SEQAN_ASSERT_EQ(getKappa(scheme),  0.041);
+        SEQAN_ASSERT_EQ(getH(scheme),      0.140);
+        SEQAN_ASSERT_EQ(getAlpha(scheme),  1.900);
+        SEQAN_ASSERT_EQ(getBeta(scheme), -30.000);
+    }
+
+    // Blosum45
+    {
```

Comment:
But all ther numbers are actually different? How would you have refactored this?